test/functional/apps/discover/_search_on_page_load.ts

```diff
@@ -76,6 +76,8 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         expect(await testSubjects.exists('refreshDataButton')).to.be(true);
         await retry.waitFor('number of fetches to be 0', waitForFetches(0));
 
+        // wait for debounce reload
+        await PageObjects.common.sleep(100);
         await testSubjects.click('refreshDataButton');
```

Comment:
Looking at it I'm not sure what happened when it failed. Since the `refreshDataButton` was already present, what could cause the click to be ignored?

Reply:
On initial load we are triggering fetch anyway [**here**](https://github.com/elastic/kibana/blob/ad6b687dadd2d98c8cd89222cdd9134edbbde1af/src/plugins/discover/public/application/main/utils/use_discover_state.ts#L225). For that specific case, when `searchOnPageLoad` is set to `false`, we are [**skipping it**](https://github.com/elastic/kibana/blob/ad6b687dadd2d98c8cd89222cdd9134edbbde1af/src/plugins/discover/public/application/main/utils/get_fetch_observable.ts#L68). So in our case we are loading page, trigger fetch and clicking `refreshDataButton` too fast, because [**debounce**](https://github.com/elastic/kibana/blob/ad6b687dadd2d98c8cd89222cdd9134edbbde1af/src/plugins/discover/public/application/main/utils/get_fetch_observable.ts#L62) will merge click event with the first fetch event into one. Then this single event will be skipped. For that reason we have to wait at least 100 sec, in this case it will be separate action.

Reply:
You can check it by changing debounce timer to 2000 msec. Open discover and click refresh button right away, it will not gonna work. You will have to wait 2 sec at least.

Reply:
thx a lot, for the context 🙏 could you add this in the comment, because it's very valid information. many thx